src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Ball {
+    private final Color color;
+    private final int number;
+
+    public Ball(Color color, int number) {
+        this.color = color;
+        this.number = number;
+    }
+
+    @Override
+    public String toString() {
+        return "Ball with color "
+                + color
+                + ", number "
+                + number
+                + ";";
+    }
```

Comment:
The toString method is correctly overridden. However, if the Color enum is customized with a toString method that could potentially be overridden, it's safer to use color.name() for getting the String representation of the enum constant to ensure that the actual enum name is used.